cli/src/action/circuit/defaults.rs

```diff
@@ -0,0 +1,73 @@
+// Copyright 2018-2020 Cargill Incorporated
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use clap::ArgMatches;
+
+use crate::error::CliError;
+use crate::store::default_value::{
+    DefaultStoreError, DefaultValue, DefaultValueStore, FileBackedDefaultStore,
+};
+
+const MANAGEMENT_TYPE_KEY: &str = "management_type";
+const SERVICE_TYPE_KEY: &str = "service_type";
+
+use super::Action;
+
+pub struct SetDefaultValueAction;
+
+impl Action for SetDefaultValueAction {
+    fn run<'a>(&mut self, arg_matches: Option<&ArgMatches<'a>>) -> Result<(), CliError> {
+        let args = arg_matches.ok_or_else(|| CliError::RequiresArgs)?;
+
+        let name = match args.value_of("name") {
+            Some(key) => key,
+            None => return Err(CliError::ActionError("name is required".into())),
+        };
+
+        let key = match name {
+            "service-type" => SERVICE_TYPE_KEY,
+            "management-type" => MANAGEMENT_TYPE_KEY,
+            _ => return Err(CliError::ActionError("Invalid default name".into())),
```

Comment:
This should probably be `format!("{} is not a valid default name", name)`

Reply:
Likewise  in the other places used.

Reply:
updated